R/drive_list.R

```diff
@@ -0,0 +1,162 @@
+#' List files on Google Drive
+#'
+#' @param path character vector, path name(s) where the Google drive files are
+#'   that you would like to list. Defaults to the "My Drive" directory.
+#' @param pattern character vector, regular expression(s) of title(s) of
+#'   documents to output in a tibble. If it is `NULL` (default), information
+#'   about all documents in drive will be output in a tibble.
+#' @param ... name-value pairs to query the API
+#' @param verbose logical, indicating whether to print informative messages
+#'   (default `TRUE`)
+#'
+#'   This will default to the most recent 100 files on your Google Drive. If you
+#'   would like more than 100, include the `pageSize` parameter. For example, if
+#'   I wanted 200, I would run `drive_ls(pageSize = 200)`.
+
+#' Helpful links for forming queries:
+#'   * <https://developers.google.com/drive/v3/web/search-parameters>
+#'   * <https://developers.google.com/drive/v3/reference/files/list>
+
+#' @return tibble containing the name, type, and id of files on your google
+#'   drive (default 100 files)
+#' @examples
+#' \dontrun{
+#' ## list user's Google Sheets
+#' drive_list(q = "mimeType='application/vnd.google-apps.spreadsheet'")
+#' }
+#'
+#' @export
+drive_list <- function(path = NULL, pattern = NULL, ..., verbose = TRUE){
+
+  folder <- NULL
```

Comment:
Looping back after reading all the path-processing stuff ... ok so now I do advocate moving all this logic into a helper function. It's the internal helper that takes a path and returns id for leafmost directory. Input is `path` (and possibly `...`? IDK). Output is something that needs to be incorporated into `q`.